### ISSUE TITLE
Return pair/tuple gives warning if putting them in a structured binding with const.

### ISSUE THREAD

**Issue by HSNB** [lefticus/cpp_weekly#330]
I found this warning C26478 unintuitive, especially when returning a std::pair or std::tuple from a function and placing it in a const structured binding (without const there's no warning). 

The reason to put it const is because I don't want to edit the values anymore. However, because of this warning and wanting to avoid it, I'm forced to not use const, or use a & reference but doing that is not intuitive when the pair or tuple is returned as a value from a function. See below.

> C26478: Don't use std::move on constant variables. (es.56).

```
std::pair<int,int> test = { 1, 2 };
const auto [ a, b ] = test;  // <-- C26478
```

```
std::pair<int,int> test = { 1, 2 };
const auto & [ a, b ] = test;  // because of reference, no warning
```

To expand to a more realistic situation where you would intuitively use "const auto [ ... ]" without a reference.

```
std::pair<int,int> some_function()
{
    return { 1, 2 };
}

... somewhere else:

const auto [ a, b ] = some_function();  // WARNING, but this is what you'd expect is the right way to code...
const auto & [ a, b ] = some_function();  // NO WARNING, but this is not how you'd expect to code when receiving a value

auto [ a, b ] = some_function();  // btw, also no warning, but the whole reason I want to make a and b constant is to avoid accidentally editing it further down the code.
```

That's really weird because you wouldn't usually code `const auto & value` when receiving a value from a function. For example, when using simply an int:

```
int some_function()
{
    return 1;
}

... somewhere else:

const auto & value = some_function();  // no one writes code like this to receive an int from a function
```

Of course, I don't just put simple ints or basic types in the pair. I also have situations where I have `std::pair<std::vector,int>`, for example, and I want to avoid at all costs doing a unnecessary copy of the vector in the pair when it's returned from the function.

I noticed this warning only started appearing in Visual Studio 2022 btw, I believe it didn't show in 2019 when I coded this originally.

Any ideas what's going on? I've been coding C++ for a some time now and have been vehemently using const on variables that I don't want edited so this is quite an icky situation. But for now I'm just using without const to avoid having the warning.

Anyway, after writing this issue here on Github I realized probably I should be doing `const auto & [ a, b ] = some_function();` for best performance and avoiding an unnecessary copy of a vector if that type of object is in the returned pair. Even tho it feels icky, probably the returned std::pair lives on the stack in the current scope even tho it "disappeared" into the structured binding, so it should be fine.

Thanks for reading! <3

**Comment by LB--** [comment]
This seems like a bug that you should submit to the Visual Studio team, not this repository.

**Comment by rolivav** [comment]
hey did you report this bug to Visual Studio? could you link the bug ticket here?

**Comment by rolivav** [comment]
Nevermind, seems to be fixed in latest VS2022 17.8 https://developercommunity.visualstudio.com/t/Wrong-C26478-warning-when-using-structur/10417562
